src/main/java/org/sonarlint/intellij/config/project/AddEditExclusionDialog.java

```diff
@@ -0,0 +1,232 @@
+      true, false, false);
+    if (project != null) {
+      fileChooser.setRoots(project.getBaseDir());
+    }
+    fileTextField.addBrowseFolderListener("Select File to Exclude",
+      "Select the file which will be excluded from SonarLint analysis",
+      project, fileChooser);
+
+    FileChooserDescriptor directoryChooser = FileChooserDescriptorFactory.createSingleFolderDescriptor();
+    if (project != null) {
+      directoryChooser.setRoots(project.getBaseDir());
+    }
+    directoryTextField.addBrowseFolderListener("Select Directory to Exclude",
+      "Select the directory which will be excluded from SonarLint analysis",
+      project, directoryChooser);
+
+    DocumentListener docListener = new DocumentAdapter() {
+      protected void textChanged(final DocumentEvent e) {
+        updateOk();
+      }
+    };
+
+    fileTextField.getTextField().getDocument().addDocumentListener(docListener);
+    directoryTextField.getTextField().getDocument().addDocumentListener(docListener);
+    globTextField.getDocument().addDocumentListener(docListener);
+
+    ActionListener listener = e -> updateControls();
+    directoryRadioButton.addActionListener(listener);
+    globRadioButton.addActionListener(listener);
+    fileRadioButton.addActionListener(listener);
+    updateControls();
+  }
+
+  @NotNull
+  @Override
+  protected Action[] createActions() {
+    return new Action[] {getOKAction(), getCancelAction(), getHelpAction()};
+  }
+
+  @Override
+  protected void doHelpAction() {
+    HelpManager.getInstance().invokeHelp("ignoreUnversionedFilesDialog");
+  }
+
+  private void updateControls() {
+    directoryTextField.setEnabled(directoryRadioButton.isSelected());
+    globTextField.setEnabled(globRadioButton.isSelected());
+    fileTextField.setEnabled(fileRadioButton.isSelected());
+    helpLabel.setEnabled(globRadioButton.isSelected());
+
+    updateOk();
+  }
+
+  private void updateOk() {
+    boolean valid = StringUtils.trimToNull(getSelectedText().getText()) != null;
+    myOKAction.setEnabled(valid);
+  }
+
+  private JTextField getSelectedText() {
+    if (directoryRadioButton.isSelected()) {
+      return directoryTextField.getTextField();
+    } else if (fileRadioButton.isSelected()) {
+      return fileTextField.getTextField();
+    } else {
+      return globTextField;
+    }
+  }
+
+  @Nullable
+  protected JComponent createCenterPanel() {
+    return panel;
+  }
+
+  @CheckForNull
+  public ExclusionItem getExclusion() {
+    if (directoryRadioButton.isSelected()) {
+      return new ExclusionItem(DIRECTORY, relative(directoryTextField.getText()));
+    } else if (fileRadioButton.isSelected()) {
+      return new ExclusionItem(FILE, relative(fileTextField.getText()));
+    } else {
+      return new ExclusionItem(GLOB, relative(globTextField.getText()));
+    }
+  }
+
+  public void setExclusion(@Nullable ExclusionItem item) {
+    setTitle("Edit SonarLint File Exclusion");
+
+    if (item != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2589)